protocol/bc/nonce.go

```diff
@@ -42,3 +47,27 @@ func (n *Nonce) SetAnchored(id Hash, entry Entry) {
 	n.Witness.AnchoredID = id
 	n.Anchored = entry
 }
+
+func (n *Nonce) checkValid(vs *validationState) error {
```

Comment:
I'm considering your suggestion. I'm not opposed, but I do like the fact that the current pattern places each `checkValid` close to the definition of the data structure it's checking. Of course that will change when each entry type gets defined in a .proto file instead of a .go file.

> composing them (package txbuilder)

Note that when we retire the legacy data structures and therefore no longer have `MapTx`, we will need a new `protocol/bc`-level mechanism for composing txentry graphs. It will probably have a builder-like interface, which will blur the distinction between `protocol/bc` and `core/txbuilder`.